packages/validator/src/utils/action.ts

```diff
@@ -0,0 +1,36 @@
+import { insertValue, replaceParamIndexWithName, replacePathsWithValues } from './utils';
+import { Roots } from '../types';
+
+export function execute(specs: any, template: any, currentPath: string[], roots: Roots) {
+  for (const action of template) {
+    let targetPath;
+
+    try {
+      targetPath = JSON.parse(
+        (action['__insert'] || action['__copy'])['__target'].replace(/(?<!\\)[']/g, '"').replace(/\\'/g, "'")
```

Comment:
I agree with @amarthadan previous suggestion to make constants out of the regexes. If the regex was a constant variable It would be easier to understand what it does.

Reply:
Also, a comment above the regex would be nice.

Reply:
Done in fae0353b9739d359b219e25b9be72f72960c1dde